msal/broker.py

```diff
@@ -0,0 +1,198 @@
+"""This module is an adaptor to the underlying broker.
+PyMsalRuntime is the package providing broker's functionality.
+
+PyMsalRuntime currently supports these Windows versions, listed in this MSFT internal link
+https://github.com/AzureAD/microsoft-authentication-library-for-cpp/pull/2406/files
+"""
+from threading import Event
+import json
+import logging
+import uuid
+
+import pymsalruntime  # ImportError would be raised on unsupported platforms such as Windows 8
+    # Its API description is available in site-packages/pymsalruntime/PyMsalRuntime.pyi
+
+logger = logging.getLogger(__name__)
+
+
+class RedirectUriError(ValueError):
+    pass
+
+
+class _CallbackData:
+    def __init__(self):
+        self.signal = Event()
+        self.auth_result = None
+
+    def complete(self, auth_result):
+        self.signal.set()
+        self.auth_result = auth_result
+
+
+def _convert_error(error, client_id):
+    context = error.get_context()  # Available since pymsalruntime 0.0.4
+    if "AADSTS50011" in context:  # In WAM, this could happen on both interactive and silent flows
+        raise RedirectUriError(  # This would be seen by either the app developer or end user
+            "MsalRuntime won't work unless this one more redirect_uri is registered to current app: "
+            "ms-appx-web://Microsoft.AAD.BrokerPlugin/{}".format(client_id))
+    return {
+        "error": "broker_error",  # Note: Broker implies your device needs to be compliant.
+            # You may use "dsregcmd /status" to check your device state
+            # https://docs.microsoft.com/en-us/azure/active-directory/devices/troubleshoot-device-dsregcmd
+        "error_description": "{}. Status: {}, Error code: {}, Tag: {}".format(
+            context,
+            error.get_status(), error.get_error_code(), error.get_tag()),
+        }
+
+
+def _read_account_by_id(account_id):
+    """Return the callback result which contains the account or error"""
+    callback_data = _CallbackData()
+    pymsalruntime.read_account_by_id(
+        account_id,
+        "correlation_id",
```

Comment:
>correlation_id [](http://example.com/codeflow?start=9&length=14)

nit, this should be a guid format and passed from the app.

Reply:
Which is similar as what you did in the other operation functions:
correlation_id or _get_new_correlation_id(),

Reply:
Nice catch! Pushing in a new commit. Resolving here.